tests/pipelines/deepfloyd_if/test_if.py

```diff
@@ -58,7 +58,7 @@ def get_dummy_inputs(self, device, seed=0):
     def test_save_load_optional_components(self):
         self._test_save_load_optional_components()
 
-    @unittest.skipIf(torch_device != "cuda", reason="float16 requires CUDA")
+    @require_non_cpu
```

Comment:
why do we make this change? float16 does not work with any accelerator, no?

Reply:
okay, we have 2 options here:

1. we allow these tests only to run on cuda or xpu. 
2. we allow them to run on all accelerators and those failed accelerators should either support it or skip these test.

I think the 2 option is better because float16 is a common date type and failing will make people consider implementing it rather than ignoring it. 

but I am fine with both. just let me know what you would prefer.  Thx!

Reply:
@faaany Let's go with Option 1 please. I feel like with 2, you just end up with noise in the CI when running on accelerators that don't support FP16. 

```suggestion
    @unittest.skipIf(torch_device not in ["cuda", "xpu"], reason="float16 requires CUDA or XPU")
    @require_accelerator
```